File: coj.uci.cu/Elections.java
/*
	By: facug91
	From: http://coj.uci.cu/24h/problem.xhtml?abb=2552
	Name: Elections
	Number: 2552
	Date: 22/10/2013
*/

import java.util.Collections;
import java.util.LinkedList;
import java.util.Scanner;

public class Elections {

	public static void main (String[] args) {
		Scanner stdin = new Scanner(System.in);
		
		int n = stdin.nextInt();
		int[] votes = new int[n];
		int total = 0;
		
		for (int i=0; i<n; i++) {
			votes[i] = stdin.nextInt();
			total += votes[i];
		}
		
		LinkedList<Double> porcentajes = new LinkedList<Double>();
		
		for (int i=0; i<n; i++) {
			porcentajes.add((votes[i] * 100.0 / total));
		}
		
		Collections.sort(porcentajes);
		
		if ((porcentajes.getLast() >= 45.0) || ((porcentajes.getLast() >= 40.0) && (porcentajes.removeLast() - 10.0 >= porcentajes.getLast()))){
			System.out.println("1");
		} else {
			System.out.println("2");
		}
		
		stdin.close();
	}
}

File: problems_solved/coj.uci.cu/Elections.cpp
/*
	By: facug91
	From: http://coj.uci.cu/24h/problem.xhtml?abb=2552
	Name: Elections
	Number: 2552
	Date: 11/07/2014
*/

#include <iostream>
#include <cstdio>
#include <cstdlib>
#include <string>
#include <cstring>
#include <cmath>
#include <algorithm>
#include <vector>
#include <queue>
#include <deque>
#include <set>
#include <map>
#include <iterator>
#include <utility>
#include <list>
#include <stack>
#include <iomanip>
#include <bitset>

#define MAX_INT 2147483647
#define MAX_LONG 9223372036854775807ll
#define MAX_ULONG 18446744073709551615ull
#define MAX_DBL 1.7976931348623158e+308
#define EPS 1e-9
#define _log2(x) log(x) * 1.44269504088896340736
//const long double PI = 2*acos(0);

#define INF 1000000000

using namespace std;

int n, votes[15], total;

int main () {
	
	int i, j;
	
	scanf("%d", &n);
	
	total = 0;
	for (i=0; i<n; i++) {
		scanf("%d", votes+i);
		total += votes[i];
	}
	
	sort(votes, votes+n, greater<int>());
	
	if (votes[0]*100 >= 45*total) printf("1\n");
	else if ((votes[0]*100 >= 40*total) && (votes[0]*100-votes[1]*100>=10*total)) printf("1\n");
	else printf("2\n");
	
	return 0;
}
